include/ast/type.h

```diff
@@ -206,5 +206,38 @@ class GlobalType {
   /// @}
 };
 
+class TagType {
+public:
+  TagType() = delete;
+  TagType(const AST::FunctionType &FType) noexcept : Type(FType) {}
+  // Getter of FunctionType
+  FunctionType &getFunctionType() noexcept { return Type; }
+  const FunctionType &getFunctionType() const noexcept { return Type; }
+
+  // Getter of the size of value that is associated with the tag
+  size_t getAssocValSize() const noexcept {
+    return Type.getParamTypes().size();
+  }
+
+private:
+  FunctionType Type;
+};
+
+class Tag {
```

Comment:
> TagSection: vector of TagType
> TagType: 0x00 (uint8_t in current), TypeIndex (uint32_t), FunctionType*
> At the end of the module loading, find the pointer of function types and set into the content of tag section.
> TagInstanance: TagType

Good idea!
One last concern, would it be better to set the pointer of function types to the tag type on validation phase? Since we have to validate the type idx of a tag to set the pointer of function types.